### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by Polfo** [Laravel-Backpack/CRUD#4203]
# Bug report

### What I did

Model:

    public function songs()
    {
        return $this->belongsToMany(\App\Models\Song::class)->withPivot('pivot_info')->withTimestamps();
    }

CrudController:

    CRUD::addField([
        'name' => 'songs',
        'type' => 'relationship',
        'attribute' => 'naam', // foreign key attribute that is shown to user // computed getter
        'subfields' => [
            [
                'name' => 'pivot_info',
                'type' => 'text',
            ]
        ],
    ]);

### What I expected to happen

On an update screen, the selection span allows to select a song in a list

### What happened

The list does not contain a list of 'naam' of each song, but empty spans instead (illustrated by the blue line in image).
![image](https://user-images.githubusercontent.com/10303392/154113560-f231f123-371a-4139-a387-3f9bf8ebe4b1.png)

### What I've already tried to fix it

Add a 'title' field in the songs database. This solves the issue.
The content or type of the 'title' field does not seem important.

### PHP VERSION:
PHP 8.0.2 (cli) (built: Feb  3 2021 18:36:40) ( ZTS Visual C++ 2019 x64 )
Copyright (c) The PHP Group
Zend Engine v4.0.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.2, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v9.0.2@4c7cd8c4e95d161c0c6ada6efa0a5af4d0d487c9

### BACKPACK VERSION:
5.0.6@2be4ff8407259f528a3541f0569ce156bab63165

**Comment by pxpm** [comment]
Hello @Polfo. Thanks for the report.

I guess you are talking about the pivot select in a n-n with pivot fields right ?

Please use `'pivotSelect' => ['attribute' => 'title']` when defining the field. Check docs: https://backpackforlaravel.com/docs/5.x/crud-how-to#belongstomany-n-n-relationship 

Thanks, I am going to close this as I don't think there is any bug here.

Please re-open if I am wrong, I was wrong alot of times before 🙃 

Best,
Pedro

**Comment by Polfo** [comment]
We received a mail from Cristian
https://backpackforlaravel.com/articles/news/backpack-v5-is-here-and-it-s-open-core?utm_source=digest&utm_medium=email
It states 

> out-of-the-box, [the subfields feature inside the relationship field](https://backpackforlaravel.com/docs/5.x/release-notes#better-than-ever-relationship-field-now-with-subfields-too) will allow you to create sub-entries right inside your Create/Update form... and change attributes on the pivot table; for every Eloquent relationship type, we have the most common 1-2 use cases covered; this alone will save you dozens of hours;

This motivated me to follow the instructions on 
https://backpackforlaravel.com/docs/5.x/crud-fields#relationship-pro
and in particular  `// 'attribute' => "title", // attribute on model that is shown to user`
and also

    'pivotSelect'=> [
        // 'ajax' => true,
        'placeholder' => 'Pick a company',
        'wrapper' => [
            'class' => 'col-md-6',
        ],
    ],

These instructions do not mention the different instructions in the crud how to, which I discovered thanks to this PR and find confusing, with two places containing similar information.

I think it is a bug if following these instructions do not work.

It is an n-n relationship with pivot indeed, I (naively?) tought it was clear from the `belongsToMany` shown in the model.

**Comment by Polfo** [comment]
Indeed, I tested it. To have this work 

        'attribute' => 'selector', // foreign key attribute that is shown to user 

I have to wrap it in `pivotSelect` like this

    'pivotSelect' => [
        'attribute' => 'selector', // foreign key attribute that is shown to user
    ],

Now I don't need the workaround of adding an unused `title` column in the database any more.
Please document that clearly https://github.com/Laravel-Backpack/docs/issues/320

**Comment by Polfo** [comment]
Thank you for all your explanation and your apologies, no need.

> At the end, it does not show the attribute example, but it says you can configure it like a regular select,

I read that back then, but did not make the link to `attribute` as it is documented differently higher up.

**Comment by pxpm** [comment]
I agree, we will try to add some more info, specially in this case, since it "automatically" does something for you (creates the pivot field for selection), we should probably give it more focus. I think the use case is better explained in the `general relationship docs` than in the `relationship field docs`, so it's probably better if we just remove duplicate examples from the `relationship field docs` section, and use only the `general relationships docs` to talk about anything related to relationship fields. I think that would avoid any kind of confusion.

Indeed, I was one of the people that wanted this feature soooo badly for so long, but I also known beforehand that it would need refactor in ALOT of places, that would take ALOT of time and energy!

I am happy that we finally did it, in terms of time saving it's HUGE, I just don't want to remember that I needed to overwritte alot of stuff in controller, writte complicated validation for repeatable ... nah!! That's past! 🙏 

Thanks for the feedback 👍
